engine/xml-config/src/main/resources/job.xsd

```diff
@@ -156,37 +165,48 @@
 		<attribute name="name" type="string" use="optional" />
 	</complexType>
 
-	<complexType name="transformerDescriptorType">
+	<complexType name="descriptorType">
 		<attribute name="ref" type="token" use="required" />
 	</complexType>
 
-	<complexType name="filterType">
+	<complexType name="outputDataStreamType">
 		<sequence>
-			<element name="descriptor" type="ab:filterDescriptorType"
-				minOccurs="1" maxOccurs="1" />
-			<element name="metadata-properties" type="ab:metadataProperties"
-				minOccurs="0" maxOccurs="1" />
-			<element name="properties" type="ab:configuredPropertiesType"
-				minOccurs="0" maxOccurs="1" />
-			<element name="input" type="ab:inputType" minOccurs="0"
-				maxOccurs="unbounded" />
-			<element name="outcome" type="ab:outcomeType" minOccurs="1"
-				maxOccurs="unbounded" />
+			<element name="job" type="ab:jobType" minOccurs="1"
+				maxOccurs="1" />
 		</sequence>
-		<attribute name="requires" type="token" use="optional">
-			<annotation>
-				<documentation>
-					The ID of a filter outcome that should function as a
-					requirement for execution of this component. Requirements are
-					applied transitively to successive components. Use the keyword
-					'_all_' to override transitive behaviour and don't apply any
-					requirements
-				</documentation>
-			</annotation>
-		</attribute>
+		<attribute name="id" type="string" use="required" />
 		<attribute name="name" type="string" use="optional" />
 	</complexType>
 
+	<complexType name="transformerType">
+		<complexContent>
+			<extension base="ab:componentType">
+				<sequence>
+					<element name="output" type="ab:outputType" minOccurs="1"
+						maxOccurs="unbounded" />
+				</sequence>
+			</extension>
+		</complexContent>
+	</complexType>
+
+	<complexType name="filterType">
+		<complexContent>
+			<extension base="ab:componentType">
+				<sequence>
+					<element name="outcome" type="ab:outcomeType" minOccurs="1"
+						maxOccurs="unbounded" />
+				</sequence>
+			</extension>
+		</complexContent>
+	</complexType>
+
+	<complexType name="analyzerType">
```

Comment:
Hmm yeah but I sort of liked to have a dedicated type here in case anything particular would show up... And anyways there's a lot of parsing code that is unfortunately NOT general so it needs to know that this type is an analyzer not just any component type.